### DISCUSSION TITLE
XRDP - Auto Login

### DISCUSSION THREAD

**Discussion by Aleex104** [neutrinolabs/xrdp#3046] [Q&A] [answered]
Hello! I have a couple of doubts, I have no idea if it can be done or not, but if it can I would love to find a way (I'm connecting from Windows to an Ubuntu 22.04).
The situation is as follows: I tried both the RDP and the VNC protocol and same result, I installed the gnome extension and it didn't work, would it be possible to modify the XRPD to allow autologin? I wanted to answer the following questions:
 
1. I can connect without unblocking a user's session, i.e. turning on the computer, not logging in and not touching anything. And remotely being able to connect, is it possible?
2. Can I make it so that logging out of the XRDP session will log me out of the computer?

Thanks in advance!

**Comment by matt335672** [comment]
@Aleex104 - I don't fully understand your question. it sounds like you are trying to access an existing desktop session over RDP. Is that right?

**Reply by Aleex104** [reply]
I've already achieved the primary objective, but now I've encountered the next issue. When someone is logged in and I attempt to connect, it doesn't allow me to do so. What I aim to accomplish is similar to the behavior of Windows Remote Desktop connection. In such cases, when attempting to connect from one Windows machine to another, an alert notifies me that the target machine is already in use. I'm then prompted to confirm whether I wish to proceed with the connection. If I confirm, the current user on the target machine receives a message asking whether they accept the connection or not. If they accept, I'm able to connect to the computer, and it logs out the previous user.

So, what I'm primarily trying to achieve using XRDP is the ability to connect to a computer, log out the current user, and allow another user to connect (I omit the confirmation part for now; I only want the disconnection of the current user and the connection of the other). It's worth noting that both users are connecting remotely but using the same account.

I'm using the translator to explain myself as best I can, sorry if you don't understand it well.

**Comment by matt335672** [comment, marked as answer]
That's fine @Aleex104 - I understand where you are now.

At the moment that isn't a use-case we support.

The best you could do is detect the a user logged in in `/etc/xrdp/startwm.sh` (using loginctl) and then kill the session if they are. You'd have to set that up yourself.

**Comment by hitjethva** [comment]
It sounds like you want to achieve two main things with your remote desktop setup:

1. Enable automatic login so that you can remotely connect to the Ubuntu system without needing to log in locally.
2. Ensure that logging out of the XRDP session also logs you out of the Ubuntu system.
Let's address each of these:

1. Automatic Login: Yes, it's possible to configure Ubuntu to automatically log in a user upon boot. This would allow you to remotely connect to the system without anyone having to manually log in locally. To set up automatic login, you can follow these steps:
Open the "Settings" application.
Navigate to the "Users" section.
Click on the user account you want to set up for automatic login.
Toggle the "Automatic Login" option to the "On" position.
You may need to enter your password to confirm the change.
After configuring automatic login, you should be able to connect remotely to the Ubuntu system without anyone needing to log in locally.

2. Logout Behavior: By default, logging out of an XRDP session does not necessarily log you out of the Ubuntu system itself. However, you can customize this behavior by configuring the XRDP session to terminate the user's session upon logout. This can be done by modifying the XRDP configuration.Here's a general outline of how you could approach this:
Locate the XRDP configuration files on your Ubuntu system. These are typically found in /etc/xrdp.
Edit the configuration file, usually named sesman.ini, using a text editor such as nano or vi.
Look for a setting related to session termination or logout behavior. This might be something like TerminateSession or LogoutScript.
Modify the setting to ensure that when a user logs out of the XRDP session, it also triggers a logout from the Ubuntu system itself.
Save the changes to the configuration file and restart the XRDP service for the changes to take effect.
Please note that the exact configuration options and steps may vary depending on your specific XRDP version and configuration. Be sure to consult the XRDP documentation or community resources for more detailed instructions.

ThinLinc provides features like session resuming, session shadowing, and support for various operating systems. It's worth considering if you require higher performance and more advanced features for your remote desktop setup.

**Reply by lcniel** [reply]
This looks like ChatGPT-generated spam.
